distributed/tests/test_steal.py

```diff
@@ -1350,3 +1350,36 @@ def test_steal_worker_state(ws_with_running_task):
     assert "x" not in ws.tasks
     assert "x" not in ws.data
     assert ws.available_resources == {"R": 1}
+
+
+@pytest.mark.slow()
+@gen_cluster(nthreads=[("", 1)] * 4, client=True)
+async def test_steal_very_fast_tasks(c, s, *workers):
+    # Ensure that very fast tasks are allowed to be stolen
+    root = dask.delayed(lambda n: "x" * n)(
+        dask.utils.parse_bytes("1MiB"), dask_key_name="root"
+    )
+
+    @dask.delayed
+    def func(*args):
+        import time
+
+        time.sleep(0.002)
+
+    ntasks = 1000
+    results = [func(root, i) for i in range(ntasks)]
+    futs = c.compute(results)
+    await c.gather(futs)
+
+    # The following is an attempt to estimate and assert a roughly uniform data
+    # distribution
+    max_ntasks_on_worker = 0
+    for w in workers:
+        ntasks_on_worker = len(w.data)
```

Comment:
Probably not important, but there's an off-by-one error since every worker should also have the root task on it.

Reply:
This entire assertion cannot be exact due to many timing issues. Right now, the one worker with the root task is the one that pulls almost all tasks https://github.com/dask/distributed/issues/6573#issuecomment-1239413543